src/integrationtests/java/com/aws/greengrass/integrationtests/deployment/SubGroupDeploymentIntegrationTest.java

```diff
@@ -0,0 +1,352 @@
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+
+
+public class SubGroupDeploymentIntegrationTest extends BaseITCase {
+
+    private static final ObjectMapper OBJECT_MAPPER =
+            new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
+    private static final String THING_GROUP_PREFIX = "thinggroup/";
+    private static final String ROOT_GROUP_NAME = "parentGroup";
+    private static final String ROOT_GROUP_DEPLOYMENT_CONFIG = "FleetConfigWithSimpleAppv1.json";
+    private static final Map<String, String> ROOT_GROUP_SERVICE_MAP = Utils.immutableMap("SimpleApp", "1.0.0");
+
+    private Kernel kernel;
+    private DeploymentQueue deploymentQueue;
+    private Path localStoreContentPath;
+    @Mock
+    private ThingGroupHelper thingGroupHelper;
+    @Mock
+    private MqttClient mqttClient;
+
+    private Map<String, CountDownLatch> groupLatchMap;
+
+    @BeforeEach
+    void before(ExtensionContext context) throws Exception {
+        ignoreExceptionOfType(context, PackageDownloadException.class);
+        ignoreExceptionOfType(context, SdkClientException.class);
+        ignoreExceptionOfType(context, DeploymentRejectedException.class);
+
+        groupLatchMap = new ConcurrentHashMap<>();
+
+        kernel = new Kernel();
+        kernel.getContext().put(ThingGroupHelper.class, thingGroupHelper);
+        kernel.getContext().put(MqttClient.class, mqttClient);
+
+        NoOpPathOwnershipHandler.register(kernel);
+        ConfigPlatformResolver.initKernelWithMultiPlatformConfig(kernel,
+                this.getClass().getResource("onlyMain.yaml"));
+
+        // ensure deployment service starts
+        CountDownLatch deploymentServiceLatch = new CountDownLatch(1);
+        kernel.getContext().addGlobalStateChangeListener((service, oldState, newState) -> {
+            if (service.getName().equals(DEPLOYMENT_SERVICE_TOPICS) && newState.equals(State.RUNNING)) {
+                deploymentServiceLatch.countDown();
+            }
+        });
+
+        // set up device config
+        setDeviceConfig(kernel, DeviceConfiguration.DEPLOYMENT_POLLING_FREQUENCY_SECONDS, 1L);
+
+        // launch kernel
+        kernel.launch();
+        assertTrue(deploymentServiceLatch.await(10, TimeUnit.SECONDS));
+
+        // get kernel details after launch
+        localStoreContentPath = Paths.get(DeploymentTaskIntegrationTest.class.getResource("local_store_content").toURI());
+        deploymentQueue = kernel.getContext().get(DeploymentQueue.class);
+
+        // setup fss such that it could send mqtt messages to the mock listener
+        FleetStatusService fleetStatusService = (FleetStatusService) kernel.locate(FLEET_STATUS_SERVICE_TOPICS);
+        fleetStatusService.getIsConnected().set(false);
+    }
+
+    @AfterEach
+    void after() {
+        if (kernel != null) {
+            kernel.shutdown();
+        }
+    }
+
+    @Test
```

Comment:
All these tests are going to be very slow because they're all starting a new kernel and running deployments. I don't think that most of these are necessary for validating the functionality.

I'd suggest that you collapse some of these test scenarios into single test cases. 

The most critical test for me is
1. Deploy to parent group
2. Deploy to subgroup with a deployment which would conflict with the parent if the subgroup logic wasn't implemented
3. Deploy to sibling subgroup with a deployment which would conflict with the parent and the previous subgroup
4. Deploy to parent group again and verify it overrides what the subgroups brought

Reply:
Ok, no problem. I will remove all other tests then!